tests/suites/test_suite_psa_crypto_pake.function

```diff
@@ -596,22 +596,57 @@ exit:
 
 typedef enum
 {
-    INJECT_ERR_NONE = 0,
-    INJECT_ERR_UNINITIALIZED_ACCESS,
-    INJECT_ERR_DUPLICATE_SETUP,
-    INJECT_ERR_INVALID_USER,
-    INJECT_ERR_INVALID_PEER,
-    INJECT_ERR_SET_USER,
-    INJECT_ERR_SET_PEER,
-    INJECT_EMPTY_IO_BUFFER,
-    INJECT_UNKNOWN_STEP,
-    INJECT_INVALID_FIRST_STEP,
-    INJECT_WRONG_BUFFER_SIZE,
-    INJECT_VALID_OPERATION_AFTER_FAILURE,
-    INJECT_ANTICIPATE_KEY_DERIVATION_1,
-    INJECT_ANTICIPATE_KEY_DERIVATION_2,
-} ecjpake_injected_failure_t;
+    ERR_NONE = 0,
+    /* errors forced internally in the code */
+    ERR_INJECT_UNINITIALIZED_ACCESS,
+    ERR_INJECT_DUPLICATE_SETUP,
+    ERR_INJECT_INVALID_USER,
+    ERR_INJECT_INVALID_PEER,
+    ERR_INJECT_SET_USER,
+    ERR_INJECT_SET_PEER,
+    ERR_INJECT_EMPTY_IO_BUFFER,
+    ERR_INJECT_UNKNOWN_STEP,
+    ERR_INJECT_INVALID_FIRST_STEP,
+    ERR_INJECT_WRONG_BUFFER_SIZE,
+    ERR_INJECT_VALID_OPERATION_AFTER_FAILURE,
+    ERR_INJECT_ANTICIPATE_KEY_DERIVATION_1,
+    ERR_INJECT_ANTICIPATE_KEY_DERIVATION_2,
+    /* erros issued from the .data file */
+    ERR_IN_SETUP,
+    ERR_IN_SET_ROLE,
+    ERR_IN_SET_PASSWORD_KEY,
+    ERR_IN_INPUT,
+    ERR_IN_OUTPUT,
+} ecjpake_error_stage_t;
+
+/*
+ * This check is used for errors issued through wrong input parameters. The
+ * check is always performed because, in case of all valid parameters, the
+ * setup function can go on
+ */
+#define SETUP_ALWAYS_CHECK_STEP( test_function, this_check_err_stage )      \
+    status = test_function;                                                 \
+    if( err_stage != this_check_err_stage )                                 \
+    {                                                                       \
+        PSA_ASSERT( status );                                               \
+    }                                                                       \
+    else                                                                    \
+    {                                                                       \
+        TEST_EQUAL( status, expected_error );                               \
+        goto exit;                                                          \
+    }
 
+/*
+ * This check is used for errors injected explicitly. The check is conditional
```

Comment:
Hmm, the distinction between the two macros was not entirely clear to me initially. I think the bit that would have helped me get it faster is this: the first macro is used for code that may either succeed or fail, and this is why it's always called - in case of success, future steps depend on this one. So I think the fact that both comments start with "This check is used for errors" is slightly misleading, as only in the second case is an error always expected.